integrations/integrations.go

```diff
@@ -0,0 +1,201 @@
+func startInProcess(
+	svrCtx *server.Context,
+	svrCfg serverconfig.Config,
+	clientCtx client.Context,
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	cmtCfg := svrCtx.Config
+	gRPCOnly := svrCtx.Viper.GetBool(FlagGRPCOnly)
+
+	g, ctx := GetCtx(svrCtx, true)
+
+	if gRPCOnly {
+		svrCtx.Logger.Info("starting node in gRPC only mode; Monomer is disabled")
+		svrCfg.GRPC.Enable = true
+	} else {
+		svrCtx.Logger.Info("starting Monomer node in-process")
+		// Start the Monomer node
+		_, monomerCleanupFn, err := startMonomerNode()
+		if err != nil {
+			return err
+		}
+		defer monomerCleanupFn()
+
+		// Add the tx service to the gRPC router. We only need to register this
+		// service if API or gRPC is enabled, and avoid doing so in the general
+		// case, because it spawns a new local CometBFT RPC client.
+		/* if svrCfg.API.Enable || svrCfg.GRPC.Enable {
+		    clientCtx = clientCtx.WithClient(local.New(tmNode))
+		    app.RegisterTxService(clientCtx)
+		    app.RegisterTendermintService(clientCtx)
+		    app.RegisterNodeService(clientCtx, svrCfg)
+		} */
+	}
+
+	grpcSrv, clientCtx, err := StartGrpcServer(ctx, g, svrCfg.GRPC, clientCtx, svrCtx, app)
+	if err != nil {
+		return err
+	}
+
+	err = StartAPIServer(ctx, g, svrCfg, clientCtx, svrCtx, app, cmtCfg.RootDir, grpcSrv)
+	if err != nil {
+		return err
+	}
+
+	if opts.PostSetup != nil {
+		if err := opts.PostSetup(svrCtx, clientCtx, ctx, g); err != nil {
+			return err
+		}
+	}
+
+	return g.Wait()
+}
+
+// Starts the Monomer node in-process in place of the Comet node.
+func startMonomerNode() (*node.Node, func(), error) {
+	chainID := monomer.ChainID(1)
+	engineWS, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, nil, err
+	}
+	cometListener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, nil, err
+	}
+	appdb := dbm.NewMemDB()
+
+	// We obviously need to replace this with an actual Monomer Application, not
+	// a testapp.App. How can I do this?
+	app, err := testapp.New(appdb, chainID.HexBig().String())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	blockdb := dbm.NewMemDB()
+	txdb := cometdb.NewMemDB()
+	mempooldb := dbm.NewMemDB()
+
+	// Do we need this cleanup function?
+	cleanupFn := func() {
+		engineWS.Close()
+		cometListener.Close()
+		appdb.Close()
+		blockdb.Close()
+		txdb.Close()
+		mempooldb.Close()
+	}
+
+	n := node.New(
+		app,
+		&genesis.Genesis{
+			ChainID:  chainID,
+			AppState: app.DefaultGenesis(),
+		},
+		engineWS,
+		cometListener,
+		blockdb,
+		mempooldb,
+		txdb,
+		&node.SelectiveListener{
+			// TODO: Remove this field from SelectiveListener
+			OnEngineHTTPServeErrCb: func(err error) {
+				fmt.Errorf("failed to serve engine HTTP: %w", err)
+			},
+			OnEngineWebsocketServeErrCb: func(err error) {
+				fmt.Errorf("failed to serve engine websocket: %w", err)
+			},
+		},
+	)
+
+	nodeCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	env := environment.New()
```

Comment:
Need to call `env.Close`. We should pass in the environment to this function and call `env.Defer`/`env.DeferErr` instead of having a cleanup function.

Reply:
So we call `env.Close` after doing the cleanup in `env.Defer`?

Reply:
Addressed.